### PR TITLE
Added category CRUD, rack CRUD already completed

### PR DESCRIPTION
**Pull request by krutalp**

### CODE REVIEW

File: server/src/controller/categoryController.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ */
+export const updateCategory = async (req, res) => {
+    if (validate(req, res)) {
+      return res;
+    }
+  
+    const {name, description } = req.body;
+    const category = await prisma.category.update({
+      where: {
+        id,
```
Comment (ChrisWilhelm) [inline server/src/controller/categoryController.js:54]: you never got the id field from the request

File: server/src/routes/category.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "/:id",
+    param("id", "Must include id in the url params").notEmpty().isInt(),
+    validator.isCategoryId,  
+    controller.deleteCategory
+  );
+
+router.post(
+    "/update",
+    body("name", "Must include name in the request").notEmpty(),
+    param("id", "Must include id in the body").notEmpty().isInt(),
```
Comment (ChrisWilhelm) [inline server/src/routes/category.js:33]: if id is going to be a param, then you need to add /:id to the URL

File: server/src/util/categoryMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+export const isUniqueName = async (req, res, next) => {
+  const { name } = req.body;
+  const query = await prisma.category.findFirst({
+    where: {
+      name,
+    },
+  });
+  if (query !== NULL) {
```
Comment (ChrisWilhelm) [inline server/src/util/categoryMiddleware.js:19]: lowercase null

File: server/src/util/categoryMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * @param {object} next - call back function
+ */
+export const isCategoryId = async (req, res, next) => {
+    const id = parseInt(req.params.id, 10);
+    const query = await prisma.category.findUnique({
+      where: {
+        id,
+      },
+    });
+    if (query === NULL || query === undefined) {
```
Comment (ChrisWilhelm) [inline server/src/util/categoryMiddleware.js:42]: lowercase null

**Review by ChrisWilhelm — CHANGES_REQUESTED** [review]

File: server/src/routes/category.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+const body = express_validator.body;
+const param = express_validator.param;
+
+const router = express.Router();
+// Guide: https://www.prisma.io/docs/concepts/components/prisma-client/crud
+
+
+router.post(
+    "/",
+    body("name", "Must include unique name for category").notEmpty(),
```
Comment (jstelnik) [inline server/src/routes/category.js:16]: maybe also do description body comment, even though not super necessary. 

body('description', 'Description is a string').optional().isString(),

**Review by jstelnik — APPROVED** [review]

File: server/src/routes/category.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+router.delete(
+    "/:id",
+    param("id", "Must include id in the url params").notEmpty().isInt(),
+    validator.isCategoryId,  
+    controller.deleteCategory
+  );
+
+router.post(
+    "/:id",
+    "/update",
```
Comment (ChrisWilhelm) [inline server/src/routes/category.js:32]: you need to combine the urls, to be like "/update/:id" or "/:id/update"

**Review by ChrisWilhelm — CHANGES_REQUESTED** [review]

**Review by ChrisWilhelm — APPROVED** [review]

**Pull request merged**